docusaurus/docs/iOS/guides/push-notifications.md

```diff
@@ -1,3 +1,104 @@
 ---
 title: Push Notifications
----
\ No newline at end of file
+---
+
+# Push Notifications
+
+First step on a way to setting up push notifications is authentication.
+Stream supports both **Certificate-based provider connection trust (.p12 certificate)** and **Token-based provider connection trust (JWT)**. Token based authentication is the preferred way to setup push notifications. This method is easy to setup and provides strong security. You can find more on setting up authentication [here](https://getstream.io/chat/docs/php/push_ios/?language=swift).
+
+## Testing Push Notifications setup
+
+Once authentication is taken care of, you can test your setup as explained below.
+
+First of all make sure you have at least one [device associated](#managing-devices-for-testing-purposes).
+
+If the device you want to test notifications on is already added to the list, you can now test pushes:
+
+    stream chat:push:test
+
+This will do several things for you:
+
+1. Pick a random message from a channel that this user is part of
+2. Use the notification templates configured for your push providers to render the payload using this message
+3. Send this payload to all of the user's devices
+   
+## Managing devices for testing purposes
+
+For testing purposes you can manage devices using Stream CLI:
+
+### Adding devices
+    stream chat:push:device:add
+
+### Removing devices
+    stream chat:push:device:delete
+
+### Getting the list of all devices registered for pushes
+    stream chat:push:device:get
+
+## Managing devices in client code
+
+If you want for a user to receive push notifications, you need to request permissions for this. One of the most common places to do so is in `AppDelegate`'s `application(_:didFinishLaunchingWithOptions:)`, for instance like this:
+
+```swift
+func application(
+    _ application: UIApplication,
+    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
+) -> Bool {
+    UNUserNotificationCenter
+        .current() 
+        .requestAuthorization(options: [.alert, .sound, .badge]) { granted, _ in
+            print("Permission granted: \(granted)")
+        }
+    return true
+}
+```
+
+In case if the device was successfully registered for receiving push notifications, your app will receive an `application(_:didRegisterForRemoteNotificationsWithDeviceToken)` callback. At this point you should send an obtained device token to StreamChat's backend to finish the device's registration for push notifications.
+
+```swift
+func application(
+    _ application: UIApplication,
+    didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data
+) {
+    chatClient.currentUserController().addDevice(token: deviceToken) { error in
+        if let error = error {
+            // handle error
+            print(error)
+        }
+    }
+}
+```
+
+### Removing devices
```

Comment:
We need to explain that this is needed during logout (but logout only, not app exit), else the registered device will get logged out user's push notifications